app/src/main/java/ca/llamabagel/transpo/ui/search/viewholders/SearchRouteViewHolder.kt

```diff
@@ -12,7 +12,11 @@ const val SEARCH_RESULT_ROUTE_LAYOUT = R.layout.search_route
 
 class SearchRouteViewHolder(private val binding: SearchRouteBinding) : RecyclerView.ViewHolder(binding.root) {
 
-    fun bind(route: SearchResult.RouteItem) {
+    fun bind(route: SearchResult.RouteItem, onClick: (searchResult: SearchResult) -> Unit) {
         binding.route = route
+
+        itemView.setOnClickListener {
```

Comment:
OnClickListeners should be bound in the holder's initializer instead of in the binding function. There is overhead associated with creating a new listener when binding and a risk of a memory leak.